exercises/forth/example_gen.go

```diff
@@ -0,0 +1,77 @@
+// +build ignore
+
+package main
+
+import (
+	"log"
+	"text/template"
+
+	"../../gen"
+)
+
+func main() {
+	t, err := template.New("").Parse(tmpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var j js
+	if err := gen.Gen("forth", &j, t); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// The JSON structure we expect to be able to unmarshal into
+type js struct {
+	ParsingAndNumbers  section `json:"parsing and numbers"`
+	Addition           section
+	Subtraction        section
+	Multiplication     section
+	Division           section
+	CombinedArithmetic section `json:"combined arithmetic"`
+	Dup                section
+	Drop               section
+	Swap               section
+	Over               section
+	UserDefinedWords   section `json:"user-defined words"`
+}
+
+// A grouped section of test cases.
+type section struct {
+	Cases []struct {
+		Description string
+		Input       []string
+		Expected    []int
+	}
+}
+
+// template applied to above data structure generates the Go test cases
+var tmpl = `package forth
+
+// Source: {{.Ori}}
+{{if .Commit}}// Commit: {{.Commit}}
+{{end}}
+
+{{/* template for repeated caseSection */}}
+
+{{define "caseSection"}}
+{{range $y, $c := .}}{
+	{{printf "%q" $c.Description}},
```

Comment:
if this line would add the section name. not sure how easy that is though, it is looking like we would have to pass two arguments into the `caseSection` template, the first being the section name, the second being the cases, and I'm not familiar enough with templates to know whether this is doable. You could try to see if you could find a way.